mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -156,6 +156,26 @@ class KMeans private (
     this
   }
 
+  // Initial cluster centers can be provided as a KMeansModel object rather than using the
+  // random or k-means|| initializationMode
+  private var initialModel: Option[KMeansModel] = None
+
+  /** Set the initial starting point, bypassing the random initialization or k-means||
+    * The condition (model.k == this.k) must be met; failure will result in an
+    * IllegalArgumentException.
+    */
+  def setInitialModel(model: KMeansModel): this.type = {
```

Comment:
@srowen Thanks for the comments. We followed this approach  based on @mengxr 's suggestion at https://issues.apache.org/jira/browse/SPARK-8018?focusedCommentId=14571757&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14571757. Please have a look.